src/uproot/interpretation/strings.py

```diff
@@ -246,6 +327,32 @@ def final_array(
             library=library,
             branch=branch,
         )
+        if (
+            isinstance(library, uproot.interpretation.library.Awkward)
+            and self._special_case
+        ):
+            trimmed = uproot._util.trim_final(
+                basket_arrays, entry_start, entry_stop, entry_offsets, library, branch
+            )
+            if all(
+                uproot._util.from_module(x, "awkward") for x in basket_arrays.values()
+            ):
+                assert isinstance(library, uproot.interpretation.library.Awkward)
+                awkward = library.imported
+                output = awkward.concatenate(trimmed, mergebool=False, highlevel=False)
+
+            self.hook_before_library_finalize(
+                basket_arrays=basket_arrays,
+                entry_start=entry_start,
+                entry_stop=entry_stop,
+                entry_offsets=entry_offsets,
+                library=library,
+                branch=branch,
+                output=output,
+            )
+            output = library.finalize(output, branch, self, entry_start, entry_stop)
+
+            return output
```

Comment:
A short-circuit `return` at the beginning of a function can be good if the if-block it's in is short and represents an exceptional case. It lets readers ignore that exception and focus on the "normal" cases below.

Here, the if-statement is a decision between two normal cases; bracketing them with `if`-`else` (and maybe `elif`) clarifies this. This one needs an `else`.